Shared/Services/EthereumService.cs

```diff
@@ -7,18 +7,28 @@ namespace Lexplorer.Services
+            return metadataLink;
+        }
+
+        public async Task<string?> GetMetadataLink(string? tokenId, string? tokenAddress, string? contractABI, string? functionName)
+        {
+
+            var web3 = new Web3("https://mainnet.infura.io/v3/53173af3389645d18c3bcac2ee9a751c");
             try
             {
-                var contract = web3.Eth.GetContract(contractABI, "0xB25f6D711aEbf954fb0265A3b29F7b9Beba7E55d");
-                var function = contract.GetFunction("uri");
-                object[] parameters = new object[1] { tokenId };
+                var contract = web3.Eth.GetContract(contractABI, tokenAddress);
+                var function = contract.GetFunction(functionName);
+                object[] parameters = new object[1] { tokenId! };
                 var uri = await function.CallAsync<string>(parameters);
+                if (uri == null) return null;
```

Comment:
this can be achieved with adding the null-safe ? below, i.e. return uri?.remove...